supress warnings

trefzer opened the pull request:
Add parameters to supress warnings, since my monitoring is not happy with warnings !

trefzer committed:
keycloak_realm: add no_password_warning parameter

set this to true, to not display the puppet warning that
the current value cannot be checked.

trefzer committed:
keycloak_identity_provider: add no_client_secret_warning parameter

set this to true, to not display the puppet warning that
the current value cannot be checked.

trefzer committed:
keycloak_ldap_user_provider: add no_client_secret_warning parameter

set this to true, to not display the puppet warning that
the current value cannot be checked.

treydock commented:
What about the ability to suppress all warnings with a single property?  

https://github.com/treydock/puppet-module-keycloak/blob/master/lib/puppet/type/keycloak_api.rb

That allows setting things that can be accessed by all types/providers.

They are set here: https://github.com/treydock/puppet-module-keycloak/blob/master/lib/puppet/provider/keycloak_api.rb#L13-L22

You can then do like `if self.class.suppress_warnings` in instance methods.

treydock commented:
Please rebase against master branch as I've recently fixed tests.

treydock commented on spec/unit/puppet/type/keycloak_identity_provider_spec.rb:
```diff
@@ -156,13 +156,14 @@
   describe 'boolean properties' do
     # Test boolean properties
     [
+      :no_client_secret_warning,
```
The current boolean tests don't work with `parent: Puppet::Parameter::Boolean` if I had to guess since the old way kept booleans as symbols and new way uses proper booleans.

trefzer replied:
I remove it, and try again !

trefzer committed:
remove non working boolean unit test for no_client_secret_warning param

This test seems not to work with the parent: Puppet::Parameter::Boolean
setting.

treydock commented on spec/unit/puppet/type/keycloak_realm_spec.rb:
```diff
@@ -285,6 +285,7 @@
   describe 'boolean properties' do
     # Test boolean properties
     [
+      :no_password_warning,
```
This would also need to be removed or remove `parent: Puppet::Parameter::Boolean` from the property.

trefzer replied:
removed

treydock commented on spec/unit/puppet/type/keycloak_ldap_user_provider_spec.rb:
```diff
@@ -257,6 +257,7 @@
   describe 'boolean properties' do
     # Test boolean properties
     [
+      :no_bind_credential_warning,
```
This would also need to be removed or remove `parent: Puppet::Parameter::Boolean` from the property.

trefzer replied:
removed

treydock commented:
If you are able to update so I can modify your fork's branch via this PR, I can make the updates.

trefzer committed:
remove more non working boolean spec tests

trefzer commented:
@treydock I think at least one statement of your last comment should be negated ;)
feel free to take over but hopefully it will pass now with all the removes !
